website/docs/using-react-context.mdx

```diff
@@ -0,0 +1,96 @@
+---
+id: using-react-context
+title: Using React Context API
+sidebar_label: Using React Context API
+---
+
+## React Context API 
+React Context API provides a easy way to pass data through the component tree without having to pass props down 
+manually at every level. You can find more about the Context API in [React documentation](https://reactjs.org/docs/context.html).
+
+You can use the Context API with React Native Navigation. In this example, we are going to create a screen which uses
+the Counter Context.
+
+## Create a Counter context
+***TODO***: Found this on the Playground app, will need to explain why we needed to do this way.
```

Comment:
@guyca Would you be able to help me out on this one? The gist of the question is because RNN is not a single root structure, we have to work around to have the native `React Context API` working.

Reply:
Will going to be adding this info: https://github.com/wix/react-native-navigation/pull/6064/files#r398980408

> The thing is, I'm not sure why someone would want to trigger a render across all of his screens as that seems very wasteful.

This would be more of a case of global state management where the same states are shared across multiple screens.